windows使用luamake构建报了个错

```bat
G:\soluna>luamake
Failed to start git process: subprocess::spawn: (sys:3)系统找不到指定的路径。
[1/7] Compile C   build/obj/soluna_src/material_blit.obj
FAILED: [code=2] build/obj/soluna_src/material_blit.obj
cl /nologo /showIncludes -c src/material_blit.c /Fobuild/obj/soluna_src/material_blit.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibuild /I3rd/lua /I3rd /I3rd/yoga /I3rd/zlib /I3rd/zlib/contrib/minizip /D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE /D_CRT_SECURE_NO_DEPRECATE /DNDEBUG -O2 -W3 -utf-8 -experimental:c11atomics /wd4244 /wd4267 /wd4305 /wd4996 /wd4018 /wd4113 /std:c11
build\blit.glsl.h(197): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(198): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(199): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(200): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(201): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(205): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(206): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\blit.glsl.h(207): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
[2/7] Compile C   build/obj/soluna_src/material_default.obj
FAILED: [code=2] build/obj/soluna_src/material_default.obj
cl /nologo /showIncludes -c src/material_default.c /Fobuild/obj/soluna_src/material_default.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibuild /I3rd/lua /I3rd /I3rd/yoga /I3rd/zlib /I3rd/zlib/contrib/minizip /D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE /D_CRT_SECURE_NO_DEPRECATE /DNDEBUG -O2 -W3 -utf-8 -experimental:c11atomics /wd4244 /wd4267 /wd4305 /wd4996 /wd4018 /wd4113 /std:c11
build\texquad.glsl.h(321): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(322): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(323): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(324): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(325): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(326): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(327): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(328): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(332): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(333): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(334): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
[3/7] Compile C   build/obj/soluna_src/material_mask.obj
FAILED: [code=2] build/obj/soluna_src/material_mask.obj
cl /nologo /showIncludes -c src/material_mask.c /Fobuild/obj/soluna_src/material_mask.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibuild /I3rd/lua /I3rd /I3rd/yoga /I3rd/zlib /I3rd/zlib/contrib/minizip /D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE /D_CRT_SECURE_NO_DEPRECATE /DNDEBUG -O2 -W3 -utf-8 -experimental:c11atomics /wd4244 /wd4267 /wd4305 /wd4996 /wd4018 /wd4113 /std:c11
build\maskquad.glsl.h(358): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(359): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(360): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(361): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(362): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(363): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(364): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(365): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(369): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(370): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\maskquad.glsl.h(371): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
[4/7] Compile C   build/obj/soluna_src/material_quad.obj
FAILED: [code=2] build/obj/soluna_src/material_quad.obj
cl /nologo /showIncludes -c src/material_quad.c /Fobuild/obj/soluna_src/material_quad.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibuild /I3rd/lua /I3rd /I3rd/yoga /I3rd/zlib /I3rd/zlib/contrib/minizip /D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE /D_CRT_SECURE_NO_DEPRECATE /DNDEBUG -O2 -W3 -utf-8 -experimental:c11atomics /wd4244 /wd4267 /wd4305 /wd4996 /wd4018 /wd4113 /std:c11
build\colorquad.glsl.h(276): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\colorquad.glsl.h(277): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\colorquad.glsl.h(278): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
[5/7] Compile C   build/obj/soluna_src/render.obj
FAILED: [code=2] build/obj/soluna_src/render.obj
cl /nologo /showIncludes -c src/render.c /Fobuild/obj/soluna_src/render.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibuild /I3rd/lua /I3rd /I3rd/yoga /I3rd/zlib /I3rd/zlib/contrib/minizip /D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE /D_CRT_SECURE_NO_DEPRECATE /DNDEBUG -O2 -W3 -utf-8 -experimental:c11atomics /wd4244 /wd4267 /wd4305 /wd4996 /wd4018 /wd4113 /std:c11
build\texquad.glsl.h(321): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(322): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(323): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(324): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(325): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(326): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(327): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(328): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(332): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(333): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\texquad.glsl.h(334): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
[6/7] Compile C   build/obj/soluna_src/material_text.obj
FAILED: [code=2] build/obj/soluna_src/material_text.obj
cl /nologo /showIncludes -c src/material_text.c /Fobuild/obj/soluna_src/material_text.obj /EHsc /Zc:__cplusplus /O2 /Zc:inline /W3 /permissive- /MD /GL /Ibuild /I3rd/lua /I3rd /I3rd/yoga /I3rd/zlib /I3rd/zlib/contrib/minizip /D_CRT_SECURE_NO_WARNINGS /D_CRT_NONSTDC_NO_DEPRECATE /D_CRT_SECURE_NO_DEPRECATE /DNDEBUG -O2 -W3 -utf-8 -experimental:c11atomics /wd4244 /wd4267 /wd4305 /wd4996 /wd4018 /wd4113 /std:c11
build\sdftext.glsl.h(387): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(388): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(389): error C2039: "storage_buffers": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(390): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(391): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(392): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(393): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(394): error C2039: "images": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(398): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(399): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
build\sdftext.glsl.h(400): error C2039: "image_sampler_pairs": 不是 "sg_shader_desc" 的成员
3rd\sokol/sokol_gfx.h(3700): note: 参见“sg_shader_desc”的声明
ninja: build stopped: subcommand failed.

```

--- a/.github/actions/soluna/action.yml
+++ b/.github/actions/soluna/action.yml
@@ -42,33 +42,6 @@ runs:
       run: |
         git submodule update --init --recursive
       shell: bash
-    - uses: actions/checkout@v5
-      if: steps.cache.outputs.cache-hit != 'true'
-      with:
-        repository: floooh/sokol-tools-bin
-        ref: "a06f19929ff8f9824ec6dd87c21329b1f205809e"
-        path: sokol-tools-bin
-        fetch-depth: 1
-    - name: Setup sokol-shdc
-      if: steps.cache.outputs.cache-hit != 'true'
-      shell: bash
-      run: |
-        if [[ "$RUNNER_OS" == "Windows" ]]; then
-          TARGET_DIR="C:/Windows/System32"
-          SHDC_BINARY="sokol-shdc.exe"
-          FIND_PATH="bin/win32"
-        elif [[ "$RUNNER_OS" == "macOS" ]]; then
-          TARGET_DIR="/usr/local/bin"
-          SHDC_BINARY="sokol-shdc"
-          FIND_PATH="bin/osx_arm64"
-        else  # Linux
-          TARGET_DIR="/usr/local/bin"
-          SHDC_BINARY="sokol-shdc"
-          FIND_PATH="bin/linux"
-        fi
-        echo "Setting up sokol-tools from sokol-tools-bin/$FIND_PATH"
-        find sokol-tools-bin/$FIND_PATH -name $SHDC_BINARY -exec cp {} $TARGET_DIR \;
-        $SHDC_BINARY -h
     - uses: yuchanns/actions-luamake@v1
       if: steps.cache.outputs.cache-hit != 'true'
       with:

--- a/clibs/soluna/compile_shader.lua
+++ b/clibs/soluna/compile_shader.lua
@@ -1,7 +1,15 @@
 local lm = require "luamake"
 local fs = require "bee.filesystem"
 
-local shdc<const> = lm.shdc or ""
+local paths = {
+  windows = "$PATH/win32/$NAME.exe",
+  macos = "$PATH/osx_arm64/$NAME",
+  linux = "$PATH/linux/$NAME",
+}
+local shdc = assert(paths[lm.os]):gsub("%$(%u+)", {
+  PATH = tostring(lm.basedir / "bin/sokol-tools-bin/bin"),
+  NAME = "sokol-shdc",
+})
 
 local function compile_shader(src, name, lang)
   local dep = name .. "_shader"

--- a/clibs/soluna/make.lua
+++ b/clibs/soluna/make.lua
@@ -19,7 +19,6 @@ if ok then
     local output = process.stdout:read "a"
     commit = output:match "^%s*(.-)%s*$"
     process:wait()
-    print("Hash version: " .. commit)
   end
 end
 

--- a/clibs/soluna/shader2c.lua
+++ b/clibs/soluna/shader2c.lua
@@ -1,22 +1,5 @@
 local subprocess = require "bee.subprocess"
-local platform = require "bee.platform"
 local shdcexe, src, target, lang = ...
-
-local function find_executable(name)
-  local handle = io.popen("where " .. name .. " 2>nul")
-  if handle then
-    local path = handle:read("*line")
-    local success = handle:close()
-    if success and path and path ~= "" then
-      return path:gsub("%s+$", "")
-    end
-  end
-  return name
-end
-
-if shdcexe == "" then
-  shdcexe = platform.os == "windows" and find_executable("sokol-shdc.exe") or "sokol-shdc"
-end
 
 local process = assert(subprocess.spawn {
   shdcexe,

